### PR TITLE
Allow callable for vectorization over particles on SVI

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pyro-ppl/numpyro/issues/1645

### CODE REVIEW

File: numpyro/infer/elbo.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ class ELBO:
 
     :param num_particles: The number of particles/samples used to form the ELBO
         (gradient) estimators.
-    :param vectorize_particles: Whether to use `jax.vmap` to compute ELBOs over the
-        num_particles-many particles in parallel. If False use `jax.lax.map`.
-        Defaults to True.
+    :param vectorize_particles: Callable to vectorize the computation of ELBOs over
```
Comment (fehiepsi) [inline numpyro/infer/elbo.py:37]: this change breaks current users. You can allow callable and check callable first. Defaults still to True (vmap).

File: numpyro/infer/elbo.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ class ELBO:
     """
     can_infer_discrete = False
 
-    def __init__(self, num_particles=1, vectorize_particles=True):
+    def __init__(self, num_particles=1, vectorize_particles=jax.lax.map):
         self.num_particles = num_particles
         self.vectorize_particles = vectorize_particles
```
Comment (fehiepsi) [inline numpyro/infer/elbo.py:51]: maybe
```python

def _apply_vmap(fn, keys):
    return vmap(fn)(keys)
...

    self.vectorize_particles = vectorize_particles
    if callable(vectorize_particles):
        self.vectorize_particles_fn = vectorize_particles
    elif vectorize_particles is True:
        self.vectorize_particles_fn = apply_vmap
    else:
        self.vectorize_particles_fn = jax.lax.map
```

Comment (juanitorduz) [reply numpyro/infer/elbo.py:56]: cool! I added this suggestion!

**Comment by juanitorduz** [comment]
Thanks for the suggestion @fehiepsi ! I added it to the PR and added some tests.

**Review by fehiepsi — APPROVED** [review]

**Comment by fehiepsi** [comment]
Thanks, @juanitorduz!

**Pull request merged**